### DISCUSSION TITLE
Using a pre-trained ASR model with a custom vocabulary

### DISCUSSION THREAD

**Discussion by miraodasilva** [speechbrain/speechbrain#1637] [Q&A]
Hi,

First of all just wanted to say this project looks really good and works really well so congrats to all the contributors.

I have a bit of a n00b question: I want to use a pre-trained ASR model (https://huggingface.co/speechbrain/asr-transformer-transformerlm-librispeech) for speech recognition on GRID (https://zenodo.org/record/3625687). The problem is that GRID has a weird vocabulary with weird sentences such as "Set White in B Zero Please", where for example the letter B is recognized as "Bee" by the ASR model, etc. 

I was wondering how can I use this ASR model with a custom vocabulary in order to adapt it to GRID's weird vocabulary? (this is possible in the google/AWS speech to text API, for example) I am assuming this is possible with changes to the tokenizer. I looked into it a bit but I couldn't quite understand some parts since I am not well versed at all on NLP, Language models and tokenizers. Was hoping someone could help me out.

Thanks a lot in advance!

**Comment by TParcollet** [comment]
Hi there,

This is a problem that isn't THAT simple to solve. Indeed, the only correct way to do this is to finetune the model to the new dataset. Instead, I would encourage you to use the wav2vec2 en model trained on CommonVoice. The problem is that a model trained with a tokenizer is tied to this tokenizer, in the example you gave, it's not such a big deal as the letter B should be interpreted correctly by the tokenizer IF you are not using language models (indeed, language models will always try to find a word corresponding to this). You could also try to remove the LM from the decoding. Note: ASR systems trained on LibriSpeech suck, all of them.

**Comment by miraodasilva** [comment]
Hi,

Thank you so much for the reply!! I tried wav2vec2 en model trained on CommonVoice and I get somewhat similar issues: it recognizes words that are actually  somewhat reasonable predictions but don't adhere to the GRID vocabulary. We will probably end up using an ASR model finetuned on GRID.

I'm curious, why do you say "ASR systems trained on LibriSpeech suck, all of them."? Do they not generalize to real world data? If not, then I would ask why Common Voice would be better, given that it contains a similar amount of hours of speech?

Thanks a lot in advance!
